src/components/ExerciseSearchBar.tsx

```diff
@@ -57,64 +71,88 @@ const ExerciseSearchBar: React.FC<ExerciseSearchBarProps> = ({ onSelectExercise
+            setSearchOnFocus(true);
+          }}
+          onBlur={() => {
+            setSearchOnFocus(false);
+          }}
           onChangeText={(value) => {
             setSearchInput(value);
           }}
-          autoFocus
         />
+        <View className="mr-2">
+          <ExerciseFilterModal
+            onClose={(filterArr) => {
+              setFilters(new Array(...filterArr));
+            }}
+          />
+        </View>
       </View>
-      <View className="divide-y-10 invisible  h-60 w-11/12 overflow-scroll border border-slate-200 bg-slate-50">
-        <ScrollView keyboardShouldPersistTaps="always">
-          {searchInput.length <= 0 ? (
-            <View className="mx-5 ">
-              {recentlyUsedExercise.current.length > 0 ? (
-                <View>
-                  <Text className="text-l mx-1 my-3 font-light">Recently Used</Text>
-                  <TouchableOpacity
-                    accessibilityRole="button"
-                    key={'recently-used-' + recentlyUsedExercise.current}
-                    onPress={() => onPressHandler(recentlyUsedExercise.current)}
-                    className="mb-1 rounded border border-slate-200 p-3"
-                  >
-                    <Text className="text-l font-bold ">{recentlyUsedExercise.current}</Text>
-                  </TouchableOpacity>
-                </View>
-              ) : (
-                <View />
-              )}
-              <Text className="text-l mx-1 my-3 font-light ">All Exercises</Text>
-              {allExercises.map((item) => {
-                return (
-                  <TouchableOpacity
-                    accessibilityRole="button"
-                    key={item}
-                    onPress={() => onPressHandler(item)}
-                    className="mb-1 rounded border border-slate-200 p-3"
-                  >
-                    <Text className="text-l font-bold">{item}</Text>
-                  </TouchableOpacity>
-                );
-              })}
-            </View>
-          ) : (
-            <View className="mx-5 ">
-              <Text className="my-1 text-xs font-light "> Search Result:</Text>
-              {searchArray.current.map((item) => {
-                return (
-                  <TouchableOpacity
-                    accessibilityRole="button"
-                    key={item}
-                    onPress={() => onPressHandler(item)}
-                    className="mb-1 rounded border border-slate-200 p-3"
-                  >
-                    <Text className="text-l font-bold ">{item}</Text>
-                  </TouchableOpacity>
-                );
-              })}
-            </View>
-          )}
-        </ScrollView>
-      </View>
+      {hideBodyOnIdle && !searchOnFocus ? (
+        <View className="h-1 w-11/12 rounded border-t border-slate-200" />
+      ) : (
+        <View className="divide-y-10 absolute mt-20 h-60 w-11/12 overflow-scroll rounded border border-slate-200">
```

Comment:
The body of the search results still doesn't show above other components. Try putting this search bar in different places above other components and see the behavior when you set `hideBodyOnIdle` to true. Might need to set the background color to white so things don't appear underneath the results too (currently transparent)